### DISCUSSION TITLE
Does the site use the current translator from Fusion for other languages? Important bug!

### DISCUSSION THREAD

**Discussion by MieszkoP** [pfusik/cito#175] [Ideas]
Does the s[website](https://fusion-lang.org/) use the current translator from Fusion for other languages? I'm asking this question because I think I've found a bug that's probably pretty important. The following code:

```
public class JakasKlasa
{
    int x = 7;
    public void setInt!(int _int)
    {
        x = _int;
    }
    
    public int getInt()
    {
        return x;
    }
}

public class Program
{
     public static void Main()
    {
        
        JakasKlasa() a;
        JakasKlasa! b = a;
        JakasKlasa() c = a;
        
        b.setInt(2);
        c.setInt(3);
        
        Console.WriteLine("a: ");
        Console.WriteLine(a.getInt());
        Console.WriteLine("b: ");
        Console.WriteLine(b.getInt());
        Console.WriteLine("c: ");
        Console.WriteLine(c.getInt());
    }   
}
```

in **c++** it is translated in such a way that we have: 
1. an object 
2.  a pointer to it 
3. a copy of it.

**C++**
```
JakasKlasa a;
JakasKlasa * b = &a;
JakasKlasa c = a;

b->setInt(2);
c.setInt(3);
```

in **C#** it is translated in such a way that we have: 
1. an object 
2. a reference of it (class is reference type in C#)
3. and another reference.

**C#**
```
JakasKlasa a = new JakasKlasa();
JakasKlasa b = a;
JakasKlasa c = a;

b.setInt(2);
c.setInt(3);
```

in **Python** it is translated in such a way that we have: 
1. an object 
2. a reference of it (copying in Python means copying reference)
3. and another a reference of it .

**Python**
```
a: JakasKlasa = JakasKlasa()
b: JakasKlasa = a
c: JakasKlasa = a

b.set_int(2)
c.set_int(3)
```

Results: 

**C++:**
```
a:
**2**
b:
**2**
c:
3
```
**C# and Python:**
```
a:
3
b:
3
c:
3
```

**Comment by pfusik** [comment]
Answering the title question, Fusion playground is not always the HEAD of master nor the latest release, but I keep it somewhere between these two.

The bug is tracked in #176.
